pliers/extractors/text_encoding.py

```diff
@@ -0,0 +1,812 @@
+'''
+New Extractors that operate exclusively on Text stimuli and
+returns embedding via different SOTA models,
+such as: word2vec, glove, fasttext (average embedding)
+       : skipthought, doc2vec, ELMO, BERT, Deep Average n/w.
+       
+Debanjan Ghosh
+Francisco Pereira
+
+'''
+from pliers.stimuli.text import TextStim, ComplexTextStim
+from pliers.extractors.base import Extractor, ExtractorResult
+from pliers.support.exceptions import PliersError
+from pliers.support.decorators import requires_nltk_corpus
+from pliers.datasets.text import fetch_dictionary
+from pliers.transformers import BatchTransformerMixin
+from pliers.utils import attempt_to_import, verify_dependencies
+from pliers.extractors.text import TextExtractor
+
+import gensim
+import nltk
+from nltk.sentiment.vader import SentimentIntensityAnalyzer
+from nltk.corpus import stopwords
+
+import logging
+import os
+import numpy as np
+import pandas as pd
+import sys
+from six import string_types
+import gzip
+from enum import Enum   
+from gensim.scripts.word2vec2tensor import word2vec2tensor
+
+'''uncomment the following two imports (tf and hub)
+   to run DAN'''
+#import tensorflow as tf
+#import tensorflow_hub as hub 
+
+
+'''bert related helper code'''
+from pliers.extractors import bert_modeling
+from pliers.extractors import bert_tokenization
+from pliers.extractors import bert_extract_features
+
+'''skipthought related helper code '''
+from pliers.extractors import skipthoughts
+
+'''SIF related helper code'''
+from pliers.extractors import sif_data_io,sif_params,SIF_embedding
+
+''' import elmo related code from AllenAI'''
+from allennlp.commands.elmo import ElmoEmbedder
+
+
+embedding_methods = Enum('embedding_methods', 'average_embedding word2vec glove')
+
+keyedvectors = attempt_to_import('gensim.models.keyedvectors', 'keyedvectors',
+                                 ['KeyedVectors'])
+doc2vecVectors = attempt_to_import('gensim.models.doc2vec','doc2vecVectors',
+                                   ['Doc2Vec.load'])
+
+logging.getLogger('smart_open').setLevel(logging.ERROR)
+logger = logging.getLogger("text_encoding_logger") 
+
+class DirectTextExtractorInterface():
```

Comment:
The pattern here of making a base Interface class which is configurable but instantiates distinct classes for each method does not follow the current style.

What we have more commonly done is implement a base class which may not anything, or at least much, on its own, but sets the common pattern for more specific subclasses. 

A simple example is `FaceRecognitionFeatureExtractor`. This class is not imported in `__init__` because its not a complete extractor. But it is then subclassed by three extractors. When necessary, the common functions can be re-implemented by these extractors. More complex examples of this is `GoogleVisionAPIExtractor` set of extractors.